ngraph/test/backend/binary_convolution.in.cpp

```diff
@@ -0,0 +1,201 @@
+    const op::PadType auto_pad{op::PadType::EXPLICIT};
+    float pad_value = 0;
+
+    auto inputs_param = make_shared<op::Parameter>(element::from<T_IN>(), inputs_shape);
+    auto filters_const = make_shared<op::Constant>(element::u1, filters_shape, &filters[0]);
+    auto bin_conv = make_shared<op::v1::BinaryConvolution>(
+        inputs_param,
+        filters_const,
+        strides,
+        pads_begin,
+        pads_end,
+        dilations,
+        op::v1::BinaryConvolution::BinaryConvolutionMode::XNOR_POPCOUNT,
+        pad_value,
+        auto_pad);
+    auto f = make_shared<Function>(bin_conv, ParameterVector{inputs_param});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
+
+    auto input_tensor = backend->create_tensor(element::from<T_IN>(), inputs_shape);
+    copy_data(input_tensor, inputs);
+    auto result = backend->create_tensor(element::from<T_IN>(), outputs_shape);
+
+    auto handle = backend->compile(f);
+    handle->call_with_validate({result}, {input_tensor});
+    EXPECT_TRUE(test::all_close_f((outputs), read_vector<T_IN>(result), MIN_FLOAT_TOLERANCE_BITS));
+}
+
+template <typename T_IN>
+static void ConvolutionTest(const std::vector<T_IN>& inputs,
+                            const Shape inputs_shape,
+                            const std::vector<T_IN>& filters,
+                            const Shape filters_shape,
+                            const std::vector<T_IN>& outputs,
+                            const Shape outputs_shape,
+                            const Strides& strides,
+                            const CoordinateDiff& padding,
+                            const Strides& dilations)
+{
+    const CoordinateDiff pads_begin{padding};
+    const CoordinateDiff pads_end{padding};
+    const op::PadType auto_pad{op::PadType::EXPLICIT};
+
+    auto inputs_param = make_shared<op::Parameter>(element::from<T_IN>(), inputs_shape);
+    auto filters_param = make_shared<op::Parameter>(element::from<T_IN>(), filters_shape);
+    auto conv = make_shared<op::v1::Convolution>(
+        inputs_param, filters_param, strides, pads_begin, pads_end, dilations, auto_pad);
+    auto f = make_shared<Function>(conv, ParameterVector{inputs_param, filters_param});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
+
+    auto input_tensor = backend->create_tensor(element::from<T_IN>(), inputs_shape);
+    copy_data(input_tensor, inputs);
+    auto filters_tensor = backend->create_tensor(element::from<T_IN>(), filters_shape);
+    copy_data(filters_tensor, filters);
+    auto result = backend->create_tensor(element::from<T_IN>(), outputs_shape);
+
+    auto handle = backend->compile(f);
+    handle->call_with_validate({result}, {input_tensor, filters_tensor});
+    EXPECT_TRUE(test::all_close_f((outputs), read_vector<T_IN>(result), MIN_FLOAT_TOLERANCE_BITS));
+}
+
+// clang-format off
+// --------------------- 1D convolution ------------------------------------------
```

Comment:
I think we should test only for 2D since it's only supported version. It is odd that we test 1D and not 3D, and also need to ignore these tests for plugins.

Reply:
Since the reference implementation is an extension of regular Convolution operation, which supports 1D, 2D and 3D (relaying into 3D), I believe we should test also 1D and 3D to check we are not introducing any misbehavior. If there is a need for CPU plugin to support 1D or 3D in the future, at least this code should be tested so it can serve as its purpose. Hence I have moved 1D and included 3D tests into op_eval directory, only tested for interpreter backend.